src/components/tab-group/tab-group.ts

```diff
@@ -0,0 +1,65 @@
+import {Component, Input} from 'angular2/core';
+import {QueryList} from "angular2/core";
+import {ContentChildren} from "angular2/core";
+import {PortalHostDirective} from "../../core/portal/portal-directives";
+import {MdTabLabel} from "./tab-label";
+import {MdTabContent} from "./tab-content";
+import {MdTabBar} from "./tab-bar";
+import {MdTabOutlet} from "./tab-outlet";
+
+let nextId: number = 0;
+
+@Component({
+  selector: 'md-tab-group',
+  templateUrl: './components/tab-group/tab-group.html',
+  styleUrls: ['./components/tab-group/tab-group.css'],
+  directives: [PortalHostDirective],
+})
+export class MdTabGroup {
+  @ContentChildren(MdTabLabel) labels: QueryList<MdTabLabel>;
+  @ContentChildren(MdTabContent) contents: QueryList<MdTabContent>;
+  @Input() selectedIndex: number = 0;
+  private _groupId: number;
+  
+  constructor() {
+    this._groupId = nextId++;
+  }
+
+  /**
+   * Returns a unique id for each tab label element
+   * @param i
+   * @returns {string}
+   */
+  getTabLabelId(i: number) {
+    return 'md-tab-label-' + this._groupId + '-' + i;
+  }
+
+  /**
+   * Returns a unique id for each tab content element
+   * @param i
+   * @returns {string}
+   */
+  getTabContentId(i: number) {
+    return 'md-tab-content-' + this._groupId + '-' + i;
+  }
+
+  /**
+   * Increment the selected index by 1; prevent going over the number of tabs
+   */
+  nextTab() {
```

Comment:
These will most-likely be reused when tab-bar is implemented, but this isn't in this iteration.  I only left this file in because Jeremy said to leave it in.